deckhouse-controller/pkg/controller/module-controllers/release/controller.go

```diff
@@ -513,41 +512,43 @@ func (c *moduleReleaseReconciler) reconcilePendingRelease(ctx context.Context, m
 
 	if releaseUpdater.PredictedReleaseIsPatch() {
 		// patch release does not respect update windows or ManualMode
-		err = releaseUpdater.ApplyPredictedRelease(nil)
-		if errors.Is(err, updater.ErrNotReadyForDeploy) {
-			//TODO: create custom error type with additional fields like reason end requeueAfter
-			return ctrl.Result{RequeueAfter: defaultCheckInterval}, nil
-		}
-
+		err = releaseUpdater.ApplyPredictedRelease()
 		if err != nil {
-			return ctrl.Result{RequeueAfter: defaultCheckInterval}, fmt.Errorf("apply predicted release: %w", err)
+			return r.wrapApplyReleaseError(err)
 		}
 
 		modulesChangedReason = "a new module release found"
 		return ctrl.Result{}, nil
 	}
 
-	var windows update.Windows
-	if !releaseUpdater.InManualMode() {
-		windows = policy.Spec.Update.Windows
-	}
-
-	err = releaseUpdater.ApplyPredictedRelease(windows)
-	if errors.Is(err, updater.ErrNotReadyForDeploy) {
-		//TODO: create custom error type with additional fields like reason end requeueAfter
-		return ctrl.Result{RequeueAfter: defaultCheckInterval}, nil
-	}
+	err = releaseUpdater.ApplyPredictedRelease()
 	if err != nil {
-		return ctrl.Result{RequeueAfter: defaultCheckInterval}, fmt.Errorf("apply predicted release: %w", err)
+		return r.wrapApplyReleaseError(err)
 	}
 
 	modulesChangedReason = "a new module release found"
-	return c.cleanUpModuleReleases(ctx, mr)
+	return r.cleanUpModuleReleases(ctx, mr)
+}
+
+func (r *moduleReleaseReconciler) wrapApplyReleaseError(err error) (ctrl.Result, error) {
+	var notReadyErr *updater.NotReadyForDeployError
+	if errors.As(err, &notReadyErr) {
+		r.logger.Infoln(err.Error())
+		// TODO: requeue all releases if deckhouse update settings is changed
+		// requeueAfter := notReadyErr.RetryDelay()
+		// if requeueAfter == 0 {
+		// requeueAfter = defaultCheckInterval
+		// }
+		// r.logger.Infof("%s: retry after %s", err.Error(), requeueAfter)
+		return ctrl.Result{Requeue: true, RequeueAfter: notReadyErr.RetryDelay()}, nil
```

Comment:
ctrl.Result{RequeueAfter: notReadyErr.RetryDelay()}

if you fill RequeueAfter - its override Requeue: true.
Or we have possibility to RequeueAfter be zero?

Reply:
RetryDelay can be zero, but now all this logic is paused, this is a reserve for the future. Changed to defaultCheckInterval